scripts.py

```diff
@@ -225,89 +230,369 @@ def run(self, data_source_name, metadata_records, epub_directory, cover_director
+            return nodes
+
+        @property
+        def siblings(self):
+            if not self.parent:
+                return []
+            return filter(lambda c: c.name!=self.name, self.parent.children)
+
+        @property
+        def default_sibling(self):
+            if self.default:
+                return self
+            default_sibling = [s for s in self.siblings if s.default]
+            if default_sibling:
+                return default_sibling[0]
+
+        def __str__(self):
+            if self.name == 'Main':
+                return ''
+
+            nodes = list()
+            current_node = self
+            while current_node.parent:
+                nodes.insert(0, current_node.name)
+                current_node = current_node.parent
+            return '>'.join(nodes)
+
+        def set_parent(self, parent):
+            self._parent = parent
+            if parent:
+                parent.add_child(self)
+            self.set_default()
+
+        def set_default(self):
+            self.default = False
+            if (not self.default_sibling and
+                ((self.name=='Fiction' and not self.parent.name=='Main')
+                or self.name.startswith('General'))):
+                self.default = True
+
+        def add_child(self, child):
+            existing_children = [c.name for c in self.children]
+            if not child.name in existing_children:
+                self._children.append(child)
+
+        def add_path(self, branch_list):
+            current = self
+            while branch_list:
+                new_node_name = branch_list.pop(0)
+
+                existing = filter(lambda c: c.name==new_node_name, current.children)
+                if existing:
+                    current = existing[0]
+                else:
+                    new = type(self)(new_node_name, parent=current)
+                    current = new
+
+    FICTIONALITY = ['fiction', 'nonfiction']
+
+    LANGUAGES = LanguageCodes.english_names_to_three.keys()
+
     @classmethod
     def arg_parser(cls):
         parser = argparse.ArgumentParser()
         parser.add_argument(
-            'data-source',
-            help='A specific source whose Works should be exported.'
+            '-s', '--source-file',
+            help='Existing CSV file or YAML list with categories'
+        )
+        parser.add_argument(
+            '-o', '--output-file', default='output.csv',
+            help='New or existing filename for the output CSV file.'
         )
         parser.add_argument(
-            '-f', '--filename', help='Name for the output CSV file.'
+            '-d', '--datasources', nargs='+',
+            help='A specific source whose Works should be exported.'
         )
         parser.add_argument(
             '-a', '--append', action='store_true',
             help='Append new results to existing file.'
         )
         return parser
 
+    @property
+    def base_works_query(self):
```

Comment:
This property was created entirely to get around copying-and-pasting the query into the test.